### ISSUE TITLE
Treatment of (log -0.) and (sqrt -0.) differs

### ISSUE THREAD

**Issue by gambiteer** [gambit/gambit#790]
We treat `(log -0.)` and `(sqrt -0.)` somewhat differently:
```
> (log -0.)
-inf.0+3.141592653589793i
> (sqrt -0.)
-0.
```
This arises from this different ways we treat -0. in `##sqrt` and `##log`:
```
    (if (##flnegative? x)
        (macro-if-cpxnum
         (##make-rectangular 0 (##flsqrt (##fl- x)))
         (range-error))
        (##flsqrt x))
```
and
```
        (if (or (##flnan? x)
                (##not (##flnegative?
                        (##flcopysign (macro-inexact-+1) x))))
            (##fllog x)
            (negative-log))
```
I noticed this after considering this commit in racket: https://github.com/racket/racket/commit/cc40f09423a48b85c1bf7db1667027b3545fa93f

I don't really know what do do here, but it seems that we should (perhaps?) do the same thing in `log` and `sqrt`.

I looked at the CLHS and a few other places, but got little guidance.

**Comment by feeley** [comment]
So what should be done?

**Comment by gambiteer** [comment]
I looked again at the C99 spec and decided to follow Racket's example.

**Comment by gambiteer** [comment]
R7RS says explicitly that `(log -0.)=> -inf.0+3.1415926535i`, which is unfortunate, as it doesn't say anything about `(sqrt -0.)`  (and I was on that committee, blah), and Gambit doesn't do it that way now, I just set it so that `(log -0.) => -inf.0`.
Is this some place where we want to follow R7RS, or just diverge.

**Comment by feeley** [comment]
Can you trace who came up with the idea that `(log -0.)` => `-inf.0+3.1415926535i`? Perhaps it was copies from CommonLisp:

```
$ sbcl --noinform --quit --eval '(progn (print (log -0.0d0)) (fresh-line))'

#C(#.DOUBLE-FLOAT-NEGATIVE-INFINITY 3.141592653589793d0)
```

CommonLisp's numeric tower was carefully designed, so it may be the right thing.

**Comment by feeley** [comment]
and...

```
$ sbcl --noinform --quit --eval '(progn (print (sqrt -0.0d0))(fresh-line))'

-0.0d0
```

**Comment by feeley** [comment]
I don't have cmucl installed, but sbcl is derived from cmucl.

**Comment by gambiteer** [comment]
Fixed with 50bd8298e628d52d2f682103fa5976eba82f8550.
